lib/Elastica/Suggest/AbstractSuggest.php

```diff
@@ -16,9 +17,9 @@
     protected $_name;
 
     /**
-     * @var string the text for this suggestion
+     * @var string the prefix for this suggestion
      */
-    protected $_text;
+    protected $_prefix;
```

Comment:
Ok I understand ur point @Zyqsempai but I think we have mixed problems : 

1 we need to solve the skipped test by CompletionTest.php class
2 we need clarify if text has been dropped or removed (it looks like in ES that text is not just a wrapper of prefix )
3 we, also, have not implemented **regex** in [suggesters completion](https://www.elastic.co/guide/en/elasticsearch/reference/current/search-suggesters-completion.html#regex)

i propose this approach : 
1 - merging the #1213 (which just solves the tests)
2 - we should implement **regex** 
3 - text has not been deprecated and not even removed everywhere, I understand that maybe is for BC but there's nothing inside ES (branches that explains that it remains for BC). 

My suggestion is to keep **text** (until we had the evidence Elastic want to remove it) and **prefix** and add **regex**, and implement them through setParam as suggested by @ruflin which seems a more optimised solution.

So the idea is to split 1 from 2-3 which could be done together.

How does it sounds ?